### PR TITLE
Improve documentations of transform/apply functions

### PR DESCRIPTION
**Pull request by xinrong-databricks**
Improve documentations of transform/apply functions.

Some transform/apply functions will add a default index when the given UDF has return type annotated. This PR raises users' attention on that.

### CODE REVIEW

**Review by itholic — APPROVED** [review]

File: docs/source/user_guide/transform_apply.rst [hunk 1]
```diff
@@ -100,6 +100,9 @@ treated that it belongs to a new different DataFrame. See also
   :align: center
   :width: 650
 
+In case of :func:`DataFrame.koalas.apply_batch`, when the given function has the return type annotated, the original index of the DataFrame will be lost and then a default index will be attached to the result DataFrame. Please be careful about configuring default index. See also
```
Comment (ueshin) [inline docs/source/user_guide/transform_apply.rst:103]: I guess `DataFrame.apply` as well? Could you confirm?

Comment (ueshin) [reply docs/source/user_guide/transform_apply.rst:103]: I guess we should rather put this in the each function's doc?

- https://koalas.readthedocs.io/en/latest/reference/api/databricks.koalas.DataFrame.apply.html
- https://koalas.readthedocs.io/en/latest/reference/api/databricks.koalas.DataFrame.koalas.apply_batch.html

Comment (ueshin) [reply docs/source/user_guide/transform_apply.rst:103]: What about `DataFrame.koalas.transform_batch` with the UDF returning `DataFrame`?

Comment (xinrong-databricks) [reply docs/source/user_guide/transform_apply.rst]: Updated!

I added GroupBy.transform as well.

Comment (ueshin) [reply docs/source/user_guide/transform_apply.rst]: ah, then GroupBy.apply too?

Comment (xinrong-databricks) [reply docs/source/user_guide/transform_apply.rst]: Ah good catch! I was greping 'retain_index=False` so missed this one. Updated!

**Review by ueshin — APPROVED** [review]
LGTM.

**Comment by xinrong-databricks** [comment]
Thanks, merging!

**Pull request merged**